magda-web-client/src/Components/Common/DataPreviewMap.js

```diff
@@ -9,7 +9,7 @@ import Spinner from "Components/Common/Spinner";
 
 export const defaultDataSourcePreference = [
     "WMS",
-    "ESRI REST",
+    ["ESRI REST", /MapServer/],
```

Comment:
Rather than have this as an array of `string | string[]` can we maybe make this an array of `{id: string; urlRegex?: RegExp}` ? I think that makes it a bit more obvious how to add new ones.

Reply:
Updated~